### PR TITLE
Implement TypeSystemSwiftTypeRef::GetReferentType() 

### PR DESCRIPTION
**Pull request by adrian-prantl**
rdar://68171397

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp [hunk 1]
```diff
@@ -2518,7 +2518,21 @@ CompilerType TypeSystemSwiftTypeRef::GetErrorType() {
...
 CompilerType
 TypeSystemSwiftTypeRef::GetReferentType(opaque_compiler_type_t type) {
-  return m_swift_ast_context->GetReferentType(ReconstructType(type));
+  auto impl = [&]() -> CompilerType {
+    using namespace swift::Demangle;
+    Demangler dem;
+    NodePointer node = GetDemangledType(dem, AsMangledName(type));
+    if (!node ||
+        (node->getKind() != Node::Kind::Unowned &&
+         node->getKind() != Node::Kind::Unmanaged) ||
```
Comment (kastiglione) [inline lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2527]: should this handle `Weak` too?

Comment (kastiglione) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2527]: oh, there's a separate function for it?

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2527]: That surprised me too and frankly I think this API is questionable, but the SwiftASTContext implementation specifically excludes weak references!

Comment (adrian-prantl) [reply lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp:2527]: The other function in SwiftLanguageRuntime also unravels the Optional type that is expected in a weak reference.

**Pull request merged**